zanata-war/src/main/java/org/zanata/service/impl/TranslationServiceImpl.java

```diff
@@ -186,6 +192,12 @@
         validateReviewPermissionIfApplicable(translationRequests,
                 projectIteration, hLocale);
 
+        DocumentLocaleKey documentLocaleKey =
+            new DocumentLocaleKey(projectIteration.getId(),
+                sampleHTextFlow.getDocument().getId(), hLocale.getLocaleId());
```

Comment:
This `documentLocaleKey` definition could be moved down to just before `new TextFlowTargetStateEvent`